backend/src/models/base/m2m_relationship.py

```diff
@@ -0,0 +1,13 @@
+import uuid
+
+from sqlalchemy import ForeignKey, UniqueConstraint
+from sqlalchemy.orm import Mapped, mapped_column
+from backend.src.models.base.base import BaseModel
+
+
+class UserTheatricalRoleRelationship(BaseModel):
+    __table_args__ = (
+        UniqueConstraint("user_id", "theatrical_role_id", name="idx_user_theatrical_role"),
+    )
+    user_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("users.id"))
+    theatrical_role_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("theatrical_roles.id"))
```

Comment:
It would be also great to have b-tree indexes for both of this fields, but again I don't know the size of our database (at least approximately)